### ISSUE TITLE
Manage Browser Agent Configurations

### ISSUE THREAD

**Issue by AshleyPoole** [newrelic/terraform-provider-newrelic#73]
**Use Case**

the use case for using terraform to create Browser SPA in New Relic is that my customer hosts a number of customers and they want each of these customer instances to have New Relic monitoring.  It would be much easier if they can use Terraform to spin up a Browser SPA app in New Relic instead of using the UI to add new apps.  

The particular customer add about 10 to 30 customers per week so you can imagine how much simpler it would be to automate this.

To allow configuring and setup of Browser agent configurations. API documentation: https://docs.newrelic.com/docs/browser/new-relic-browser/browser-agent-spa-api
https://rpm.newrelic.com/api/explore/browser_applications/create

**Comment by jpvajda** [comment]
@glitton also identified this as a need.

**Comment by simontabor** [comment]
We have multiple environments which need the Browser configurations kept in sync. Doing this manually is an absolute nightmare and Terraform'ing them would be a huge help

**Comment by kidk** [comment]
There's some work happening to port the Browser Application API from REST to GraphQL, after which we can update Terraform to support it. No ETA for now, but we're continuing to track.

**Comment by mrytower05** [comment]
@kidk 
The interface to create a new browser agent, and get its loader script, is now live in nerdgraph. Example:

```
mutation createNewBrowser {
    agentApplicationCreateBrowser(accountId:###, name:"My New Browser App", settings:{loaderType:SPA}) {
    guid
    settings {     
      loaderScript
    }
  }
} 
```
(If you would like to enable browser reporting in an existing APM application that is instrumented via one of our language agents, you can use agentApplicationEnableApmBrowser to configure the agent with server-side config to begin injecting the browser agent - the GUID for the new application will get created as data begins to report.)

There are some additional settings available that can be passed at create time. Settings can be updated after creation with the agentAppliationSettingsUpdate mutation. The loader script can be accessed after creation via the entity field:

```
query getLoaderScript {
  actor {
    entity(guid:"<your guid here>") {
      ...on BrowserApplicationEntity {
        browserProperties {
          jsLoaderScript
        }
      }
    }
  }
} 
```

**Comment by kidk** [comment]
Thanks @mrytower05 we'll take a look

**Comment by arhill05** [comment]
Any movement on this?

**Comment by NSSPKrishna** [comment]
@arhill05, this is already planned in next sprint and we will get back to you once done.

**Comment by ChandraIragavarapu** [comment]
Released as part of 3.16.0
